### PR TITLE
added await expect for callback + removed unnecessary awaits

### PR DESCRIPTION
**Pull request by asanzgom**

### CODE REVIEW

File: lib/topic.ts [hunk 1]
```diff
@@ -3,9 +3,10 @@ import { showElementActions } from './kafka';
 import { navigateToKafkaList } from './navigation';
 
 export const navigateToKafkaTopicsList = async function (page: Page, kafkaName: string) {
-  await expect(page.getByText(kafkaName)).toHaveCount(1);
-  await page.locator('a', { hasText: kafkaName }).click();
-  await expect(await page.locator('button[aria-label="Topics"]')).toHaveCount(1);
+  await expect(page.getByText(`${kafkaName}`)).toHaveCount(1);
+  await page.locator('a', { hasText: `${kafkaName}` }).click();
+  await expect(page.locator('section[aria-label="Metrics view"]')).toBeVisible();
```
Comment (Frawless) [inline lib/topic.ts:8]: why do we need this?

File: tests/rhosak/kas_with_instance.spec.ts [hunk 2]
```diff
@@ -109,7 +107,7 @@ const filterByName = async function (page, name, skipClick = false) {
 // test_3kas.py test_kas_kafka_filter_by_name
 test('test instances can be filtered by name', async ({ page }) => {
   await filterByName(page, 'test');
-  await expect(page.getByText(testInstanceName)).toBeTruthy();
+  expect(page.getByText(testInstanceName)).toBeTruthy();
```
Comment (Frawless) [inline tests/rhosak/kas_with_instance.spec.ts:110]: is this correct? Are we sure that page is fully loaded right after click?

File: lib/topic.ts (hunk 1 above)
Comment (jcoleman0redhat0com) [reply lib/topic.ts:8]: We were waiting for just the menus to be rendered, so we would then select the topics menu.  This change means that we also wait for the content under the menus to be rendered.  Whilst it's possible to click on the menu before the content is there, the change better reflects user interaction.

Comment (Frawless) [reply lib/topic.ts:8]: Shouldn't we click on topics and wait for the rendering page there? I as a user will not wait until some metrics section in the dashboard will be visible if I want to see topics.

**Comment by kornys** [comment]
closing

**Pull request closed without merging**